_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -486,7 +495,7 @@ namespace UMC_MPEG2_DECODER
             auto umcRes = StartFrame(slice);
             if (umcRes != UMC::UMC_OK || !m_currFrame)
             {
-                delete slice;
+                m_lastSlice.reset(slice);
```

Comment:
it is not what it was previously... maybe you wanted to
```
m_lastSlice.reset();
```

Reply:
yes, it is not as before. in case of "m_lastSlice.reset();" the slice will be skipped (as without the change)

Reply:
ok